### DISCUSSION TITLE
Unique route names - Do routes need a name?

### DISCUSSION THREAD

**Discussion by SjorsO** [laravel/framework#32998] [General]
The [Laravel 7 upgrade guide](https://laravel.com/docs/7.x/upgrade#unique-route-names) says the following:
> you should always provide unique names for your routes

I'm currently updating a project that doesn't use named routes. It uses the `action()` helper instead. Do I need to give each route a unique name to prevent "unexpected behavior in multiple areas of the framework"? Or does this only apply to routes that have been given the same name?

**Comment by driesvints** [comment]
Think in your case a name will be generated for each route. This was more as an indication for route who were grouped under the same "as" prefix and didn't have specific route names. Those wouldn't be unique.
